source/common/router/upstream_request.cc

```diff
@@ -111,14 +111,24 @@ UpstreamRequest::UpstreamRequest(RouterFilterInterface& parent,
 
   // The router checks that the connection pool is non-null before creating the upstream request.
   auto upstream_host = conn_pool_->host();
+  absl::optional<Upstream::ClusterInfoConstSharedPtr> cluster_info =
+      parent_.callbacks()->streamInfo().upstreamClusterInfo();
+
   Tracing::HttpTraceContext trace_context(*parent_.downstreamHeaders());
+  Tracing::UpstreamContext upstream_context(upstream_host, // host_
+                                            cluster_info.has_value() ? cluster_info.value()
+                                                                     : nullptr, // cluster_info_
```

Comment:
Get the cluster info from the `upstream_host` directly.

Reply:
Done